crypto/param_build.c

```diff
@@ -277,25 +277,32 @@ static OSSL_PARAM *param_bld_convert(OSSL_PARAM_BLD *bld, OSSL_PARAM *param,
     return param;
 }
 
-OSSL_PARAM *ossl_param_bld_to_param(OSSL_PARAM_BLD *bld, void **secure)
+OSSL_PARAM *ossl_param_bld_to_param(OSSL_PARAM_BLD *bld, void **secure,
+                                    size_t *sec_sz)
 {
     OSSL_PARAM_BLD_BLOCK *blk, *s = NULL;
     OSSL_PARAM *param;
     const size_t p_blks = bytes_to_blocks((bld->curr + 1) * sizeof(*param));
     const size_t total = ALIGN_SIZE * (p_blks + bld->total_blocks);
+    size_t ss;
 
     if (bld->secure_blocks > 0) {
+        ss = bld->secure_blocks * ALIGN_SIZE;
+        if (sec_sz != NULL)
+            *sec_sz = ss;
         if (secure == NULL) {
             CRYPTOerr(CRYPTO_F_OSSL_PARAM_BLD_TO_PARAM,
                       CRYPTO_R_INVALID_NULL_ARGUMENT);
             return NULL;
         }
-        s = OPENSSL_secure_malloc(bld->secure_blocks * ALIGN_SIZE);
+        s = OPENSSL_secure_malloc(ss);
         if (s == NULL) {
             CRYPTOerr(CRYPTO_F_OSSL_PARAM_BLD_TO_PARAM,
                       CRYPTO_R_SECURE_MALLOC_FAILURE);
             return NULL;
         }
+    } else if (sec_sz != NULL) {
+        *sec_sz = 0;
```

Comment:
... and this, and change this (below):

``` C
    if (secure != NULL)
        *secure = s;
```

to this:

``` C
    if (secure != NULL) {
        *secure = s;
        *sec_sz = ss;
    }
```

Reply:
Mostly done.